WordPress/gutenberg
# Gallery block: turn on auto-migration of v1 Gallery blocks to v2 format when edited

## Description

Turns on automigration of v1 gallery blocks to v2 format unless a site has `use_BalanceTags`  enabled and is not running WP >= 5.9

## Testing
- **Before checking out this PR**, on a site on WP < 5.9 set use_balanceTags option to 1 
- Then check out this PR and add a gallery and check that it is still the old format (see screenshots below) when editing and viewing and when reloading in the editor
- Switch use_balanceTags to 0 and  load the v1 Gallery block in the editor and check that it auto migrates to the v2 block format (see screenshots below) and can be edited, saved, reloaded ok
- Try setting `use_balanceTags` back to 1 and you should get an error message about needing to upgrade to WP 5.9
- For both v1 and v2 blocks check that you can transform to individual Image blocks and back to Gallery block again, and when transforming back to Gallery make sure it is v1 format for `use_balanceTags` site, and v2 for others 
- Also try adding a v1 block with external images (to do this add Image blocks and use the replace button to add the url of an external image - then transform the Image blocks to a Gallery block) and make sure these migrate to v2 gallery format ok. There is a [known Image size selector error](https://github.com/WordPress/gutenberg/issues/36554) with this flow

**Important note**: It is not possible to load v2 version blocks in a site with `use_balanceTags` on and WP < 5.9, they will show an invalid error. The plan is to add some sort of explanatory error to indicate that user has to turn off `use_balanceTags` or upgrade to 5.9 in these instances - there is no point in showing a valid block as it will just break when the user saves it, but the user will not know unless they reload or view in the frontend, so better to fail on first editor load.

You can use `wp option set use_balanceTags 1` to toggle the option, or you can make the option available at `/wp-admin/options-writing.php` by running `wp option set initial_db_version 32452`

## Screenshots

v1 Gallery block:
<img width="664" alt="Screen Shot 2021-11-16 at 9 50 57 PM" src="https://user-images.githubusercontent.com/3629020/142125678-724eabc5-c9a3-4b9a-96ea-289278ccb2b3.png">

v2 Gallery block:
<img width="645" alt="Screen Shot 2021-11-16 at 9 58 45 PM" src="https://user-images.githubusercontent.com/3629020/142125871-0645fea7-7ae6-4ce0-a715-eab6f9ee8c60.png">
